addons/website_slides/models/slide_channel.py

```diff
@@ -199,12 +198,20 @@ def _compute_user_statistics(self):
         for record in self:
             record.completed, record.completion = mapped_data.get(record.id, (False, 0))
 
-    @api.depends('channel_type', 'partner_ids', 'upload_group_ids')
-    def _compute_access(self):
+    @api.depends('upload_group_ids')
+    def _compute_can_upload(self):
         for record in self:
-            is_doc = record.channel_type == 'documentation'
             record.can_upload = not self.env.user.share and (not record.upload_group_ids or bool(record.upload_group_ids & self.env.user.groups_id))
-            record.can_publish = record.can_upload and self.env.user.has_group('website.group_website_publisher') and (is_doc or record.user_id == self.env.user)
+
+    @api.depends('channel_type', 'user_id', 'can_upload')
+    def _compute_can_publish(self):
+        """ For channels of type 'training', only the responsible (see user_id field) can publish slides.
+        (Special case for sudo user also needs to be handled)"""
```

Comment:
I would add in docstring: sudo user need to be handled because he is user for upload done in front-end if user is not publisher but fulfill the  ``upload_group_ids`` condition

Reply:
Much clearer explanation, I've replaced mine with yours.